api/src/test/java/org/apache/iceberg/catalog/TestTableIdentifier.java

```diff
@@ -60,11 +60,11 @@ public void testToLowerCase() {
   public void testInvalidTableName() {
     Assertions.assertThatThrownBy(() -> TableIdentifier.of(Namespace.empty(), ""))
         .isInstanceOf(IllegalArgumentException.class)
-        .hasMessage("Invalid table name: null or empty");
+        .hasMessage("Invalid table name: empty");
 
     Assertions.assertThatThrownBy(() -> TableIdentifier.of(Namespace.empty(), null))
-        .isInstanceOf(IllegalArgumentException.class)
-        .hasMessage("Invalid table name: null or empty");
+        .isInstanceOf(NullPointerException.class)
```

Comment:
this is because the Immutables lib does an `Objects.requireNonNull(name, "name")` for everything that isn't marked as `@Nullable`

Reply:
alternatively, we could add `@Value.Style(throwForNullPointer = IllegalArgumentException.class)` to throw an IAE for nulls